netavark.spec.rpkg

```diff
@@ -58,12 +61,27 @@ popd
 %install
 %{__make} DESTDIR=%{buildroot} PREFIX=%{_prefix} install
 
+%post
+if [ $1 -eq 1 ]; then
+    %{_bindir}/systemctl enable --now %{name}-dhcp-proxy.service
+fi
+
+%preun
+%systemd_preun %{name}-dhcp-proxy.service
+
+%postun
+%systemd_postun %{name}-dhcp-proxy.service
+
```

Comment:
I am not fully understand what these macros do, looking at https://docs.fedoraproject.org/en-US/packaging-guidelines/Scriptlets/#_systemd does not really help me.

However I do not think we should enable it by default and it should not be restarted on upgrade otherwise the dhcp leases can fail.

Reply:
@Luap99 try `rpm --eval "%systemd_post netavark-dhcp-proxy.service"` which will expand the macro. Similar for %systemd_preun and %systemd_postun / %systemd_postun_with_restart.

```
$ rpm --eval "%systemd_post netavark-dhcp-proxy.service"


if [ $1 -eq 1 ] && [ -x "/usr/lib/systemd/systemd-update-helper" ]; then
    # Initial installation
    /usr/lib/systemd/systemd-update-helper install-system-units netavark-dhcp-proxy.service || :
fi
```

If we don't want it to be started on initial install and also not restarted on upgrade, we can keep the `%systemd_preun` and `%systemd_postun` parts and only remove `%post`. Restart on upgrade would be handled by `%systemd_postun_with_restart` which isn't here anyway.